### PR TITLE
Bosco install use yarn if .yarnrc exists in service folder

### PR DESCRIPTION
**Pull request by cliftonc**
Sort of dependent: https://github.com/yarnpkg/yarn/issues/606

If you add a .yarnrc with:

```
registry "http://npm.tescloud.com"
```

It works in your service.

### CODE REVIEW

**Comment by gabceb** [comment]
I was called "Mr Fancy Pants Cebrian" for bringing up graphql this morning and then this happens 😂😂

**Comment by cliftonc** [comment]
I just wanted to beat @bettiolo and @tomruttle to raising the issue to add it :D

**Comment by bettiolo** [comment]
@cliftonc the most hipster CTO ever.

File: commands/install.js [hunk 1]
```diff
@@ -12,6 +12,12 @@ module.exports = {
   description: 'Runs npm install against all repos',
   usage: '[-r <repoPattern>]',
   requiresNvm: true,
+  options: [{
+    name: 'yarn',
+    alias: 'y',
```
Comment (cliftonc) [inline commands/install.js]: Agreed.

**Comment by glasseyes42** [comment]
Could it not be a setting of the project - perhaps in the bosco-service.json?

**Comment by cliftonc** [comment]
It's probably better off as a global bosco config setting (like the AWS config), so you set it once and it sticks.  The problem is you need the .yarnrc in every project - but we could have it write one if there isn't one until they allow you to set the repo in each command like npm does.

**Comment by bettiolo** [comment]
Or you can check if there is a .yarnrc and use yarn instead of npm?

**Comment by cliftonc** [comment]
Updated as per last comment.

**Comment by Fetz** [comment]
👍

**Comment by geophree** [comment]
Does `yarn` check parent directories?  We could put a `.yarnrc` in the `resources` team git repo.

**Comment by bettiolo** [comment]
@cliftonc I added support to TES Private registry in yarn by doing:
```
yarn config set registry http://npm-test.tescloud.com
```

**Comment by bettiolo** [comment]
Maybe it would be better if we were looking for the `yarn.lock` file?

File: commands/install.js [hunk 2]
```diff
@@ -22,8 +22,12 @@ function install(bosco, progressbar, bar, repoPath, repo, next) {
   }
 
   var yarnrcFile = [repoPath, '.yarnrc'].join('/');
-  if (bosco.exists(yarnrcFile)) {
+  var yarnLockFile = [repoPath, 'yarn.lock'].join('/');
+  if (bosco.exists(yarnrcFile) || bosco.exists(yarnLockFile)) {
     var yarnComamnd = 'yarn';
+    if (bosco.config.get('npm:registry')) {
```
Comment (bettiolo) [inline commands/install.js]: Isn't it a bit dodgy to change people's yarn config?

File: commands/install.js [hunk 3]
```diff
@@ -21,23 +21,14 @@ function install(bosco, progressbar, bar, repoPath, repo, next) {
...
-    } else {
-      npmCommand = bosco.options.nvmUseDefault + 'npm';
-    }
+  var yarnrcFile = [repoPath, '.yarnrc'].join('/');
+  var yarnLockFile = [repoPath, 'yarn.lock'].join('/');
+  if (bosco.exists(yarnrcFile) || bosco.exists(yarnLockFile)) {
+    var yarnComamnd = 'yarn';
     if (bosco.config.get('npm:registry')) {
-      npmCommand += ' --registry ' + bosco.config.get('npm:registry');
+      yarnComamnd = 'yarn config set registry ' + bosco.config.get('npm:registry') + '; ' + yarnComamnd;
```
Comment (bettiolo) [inline commands/install.js]: Isn't it a bit dodgy to change people's yarn config?

**Comment by cliftonc** [comment]
Probably if either exist it should use yarn?

**Comment by bettiolo** [comment]
I would merge if we didn't automatically set people's registry.

File: commands/install.js [hunk 4]
```diff
@@ -21,23 +21,11 @@ function install(bosco, progressbar, bar, repoPath, repo, next) {
...
-    } else {
-      npmCommand = bosco.options.nvmUseDefault + 'npm';
-    }
-    if (bosco.config.get('npm:registry')) {
-      npmCommand += ' --registry ' + bosco.config.get('npm:registry');
-    }
-    npmCommand += ' install';
-
-    exec(npmCommand, {
+  var yarnrcFile = [repoPath, '.yarnrc'].join('/');
```
Comment (bettiolo) [inline commands/install.js]: Why are you not using https://nodejs.org/api/path.html#path_path_join_paths?

File: commands/install.js [hunk 5]
```diff
@@ -21,23 +21,11 @@ function install(bosco, progressbar, bar, repoPath, repo, next) {
...
+  var yarnLockFile = [repoPath, 'yarn.lock'].join('/');
+  if (bosco.exists(yarnrcFile) || bosco.exists(yarnLockFile)) {
+    var yarnComamnd = 'yarn';
+    exec(yarnComamnd, {
```
Comment (bettiolo) [inline commands/install.js]: Having a variable for that seems redundant.

**Pull request merged**